modules/Bio/EnsEMBL/Slice.pm

```diff
@@ -3759,6 +3759,27 @@ sub add_synonym{
   return;
 }
 
+=head2 feature_so_acc
+
+  Example     : $slice_so_acc = $slice->feature_so_acc;
+  Description : This method returns a string containing the SO accession number of the slice, based on the coordinate system name.
+  Returns     : string (Sequence Ontology accession number)
+=cut
+
+sub feature_so_acc {
+  my $self = shift;
+
+  my %region_so_mapping = (
+   'chromosome'  => 'SO:0000340',
+   'supercontig' => 'SO:0000148',
+   'scaffold'    => 'SO:0000148',
+   'contig'      => 'SO:0000149'
+  );
+
+  # return the region SO acc, or Slice acc
+  return %region_so_mapping{$self->coord_system_name} // 'SO:0000001';
```

Comment:
should that not be $region_so_mapping{$self->coord_system_name}?
Older versions of Perl do not seem happy with that syntax